### PR TITLE
Adding all optional and recommended fields to the viral template

### PR DESCRIPTION
**Pull request by bedroesb**
The PR adds all optional and recommended fields + it will no longer be mandatory to have values when an optional column is given. (important for when there is not always a value for each row)

TODO: adding documentation

### CODE REVIEW

File: ena_upload/ena_upload.py [hunk 1]
```diff
@@ -578,7 +578,7 @@ def process_args():
 
     parser.add_argument('--tool_version',
                         dest='tool_version',
-                        default='0.3.0',
+                        default='0.3.1',
```
Comment (bgruening) [inline ena_upload/ena_upload.py:581]: we should probably make this a variable that we can reuse and only bump in one place

File: ena_upload/templates/ENA_template_vir_sample.xml [hunk 2]
```diff
@@ -1,4 +1,12 @@
 <?xml version="1.0" encoding="UTF-8"?>
+<?python
+import pandas as pd
+def attributetest(row, column):
+    emptyvalue = [' ', 'nan', 'na']
```
Comment (bgruening) [inline ena_upload/templates/ENA_template_vir_sample.xml:5]: what about `''`

Comment (bedroesb) [reply ena_upload/templates/ENA_template_vir_sample.xml:5]: empty cells become an NaN object with pandas, which is detected with the pd.notna() function

File: ena_upload/ena_upload.py (hunk 1 above)
Comment (bedroesb) [reply ena_upload/ena_upload.py:581]: I agree, using maybe a _version.py file with `__version__ = "0.3.2"` in it?

File: ena_upload/templates/ENA_template_vir_sample.xml (hunk 2 above)
Comment (bgruening) [reply ena_upload/templates/ENA_template_vir_sample.xml:5]: ok, I'm just worried what happend when a cell contains `'    '`

File: ena_upload/ena_upload.py (hunk 1 above)
Comment (bgruening) [reply ena_upload/ena_upload.py:581]: jepp! :)

File: ena_upload/templates/ENA_template_vir_sample.xml (hunk 2 above)
Comment (bedroesb) [reply ena_upload/templates/ENA_template_vir_sample.xml:5]: When it only contains a space than that cell will be ignored and not be put in the template. Should this be different better?

Comment (bedroesb) [reply ena_upload/templates/ENA_template_vir_sample.xml:5]: Sorry that I already bumped a version, we wanted to make some improvements for Estonia as fast as possible and did tests with all the optional fields

Comment (bgruening) [reply ena_upload/templates/ENA_template_vir_sample.xml:5]: @bedroesb don't be sorry. Really this PR was great. I just wanted to add some comments, maybe for the next version.

Fast as possible is always good! Rock on!

Comment (bgruening) [reply ena_upload/templates/ENA_template_vir_sample.xml:5]: > When it only contains a space than that cell will be ignored and not be put in the template. Should this be different better?

sorry github does not render it correctly, is that also working if there are 5 spaces?

Comment (bedroesb) [reply ena_upload/templates/ENA_template_vir_sample.xml:5]: ow good catch ... no it doesn't, I found this python function `str.isspace()` that could check for it. I will add this with another version. I would like to support more sample types too in the future.

File: ena_upload/ena_upload.py (hunk 1 above)
Comment (bedroesb) [reply ena_upload/ena_upload.py:581]: I added it in #16 !

**Pull request merged**